### PR TITLE
Pylint and submodule update

### PR DESCRIPTION
**Pull request by IvarsKarpics**

### CODE REVIEW

File: gui/bricks/AttenuatorsBrick.py [hunk 1]
```diff
@@ -136,7 +135,7 @@ def current_value_changed(self):
             self.new_value_validator.validate(input_field_text, 0)[0]
             == QtImport.QValidator.Acceptable
         ):
-            api.transmission.set_value(float(input_field_text))
+            api.transmission.set_transmission(float(input_field_text))
```
Comment (rhfogh) [inline gui/bricks/AttenuatorsBrick.py:138]: OK for now, but presumably this wil be revisited as we refactor further, right?

File: gui/bricks/DuoStateBrick.py [hunk 2]
```diff
@@ -419,13 +419,15 @@ def __getstate__(self):
...
         try:
-            #Python2 
+            # Python2
             import new
+
             self.setIn = new.instancemethod(lambda self: None, self)
             self.setOut = self.setIn
             self.getState = new.instancemethod(lambda self: "unknown", self)
-        except:
+        except BaseException:
```
Comment (rhfogh) [inline gui/bricks/DuoStateBrick.py]: I propose
Except ImportError
(which is more genral than it's subtype ModuleNotFoundError)

File: gui/bricks/SlitsBrick.py [hunk 3]
```diff
@@ -52,7 +52,7 @@
 
 try:
     uni_chr = unichr
-except:
+except BaseException:
```
Comment (rhfogh) [inline gui/bricks/SlitsBrick.py]: "Except NameError" instead

File: gui/widgets/create_task_base.py [hunk 4]
```diff
@@ -79,7 +79,7 @@ def __init__(self, parent, name, fl, task_node_name="Unamed task-node"):
                 "energyLimitsChanged", self.set_energy_limits
             )
             api.transmission.connect(
-                "valueChanged", self.set_transmission
+                "transmissionChanged", self.set_transmission
```
Comment (rhfogh) [inline gui/widgets/create_task_base.py:82]: Do we have a policy on this? Or is it 'for now pending further refactoring'?

File: gui/widgets/heat_map_widget.py [hunk 5]
```diff
@@ -34,11 +34,13 @@
...
 class HeatMapWidget(QtImport.QWidget):
-    def __init__(self, parent=None, allow_adjust_size=True):
+
+    def __init__(self, parent=None, show_aligned_results=False):
 
         QtImport.QWidget.__init__(self, parent)
+
         self.setObjectName("heat_map_widget")
-        self.allow_adjust_size = allow_adjust_size
+        self.__show_aligned_results = show_aligned_results
```
Comment (rhfogh) [inline gui/widgets/heat_map_widget.py]: Do you specifically wwant to hide this attribute from subclasses? If not, should  it not be a single underscore?

File: gui/widgets/heat_map_widget.py [hunk 6]
```diff
@@ -239,13 +240,12 @@ def __init__(self, parent=None, allow_adjust_size=True):
             "Move to position", self.move_to_best_position_clicked
         )
         self._best_pos_table.contextMenuEvent = self.open_best_pos_popup_menu
-        # self._best_pos_table.setHidden(True)
 
-        screenShape = QtImport.QDesktopWidget().screenGeometry()
-        self.__heat_map_max_size = (screenShape.width() / 2, screenShape.height() / 2)
+        screen_shape = QtImport.QDesktopWidget().screenGeometry()
+        self.__heat_map_max_size = (screen_shape.width() / 2, screen_shape.height() / 2)
```
Comment (rhfogh) [inline gui/widgets/heat_map_widget.py:245]: Again, should that be a single underscore: _heat_map_max_size

File: gui/widgets/heat_map_widget.py [hunk 7]
```diff
@@ -346,26 +327,25 @@ def score_type_changed(self, score_type_index):
...
-            if self.__results_display[self.__score_key].ndim == 1:
-                self._heat_map_plot.hide_all_curves()
-                self._heat_map_plot.show_curve(self.__score_key)
-                self.refresh()
-            else:
-                self._heat_map_plot.plot_result(
-                    self.__results_display[self.__score_key]
-                )
-                self._heat_map_plot.add_colorbar()
+        if not self.__show_aligned_results:
```
Comment (rhfogh) [inline gui/widgets/heat_map_widget.py]: double underscores etc.

File: gui/widgets/xray_imaging_results_widget.py [hunk 8]
```diff
@@ -157,28 +157,26 @@ def __init__(self, parent=None, name=None, fl=0, xray_imaging_params=None):
             self.last_image_button_clicked
```
Comment (rhfogh) [inline gui/widgets/xray_imaging_results_widget.py]: Presumably it is deliberate that last_image_button_clicked still ahs a 'self.', whereas first_image_button_clicked does not?

**Review by rhfogh — CHANGES_REQUESTED** [review]
I approve merging, in general, though I have neither checked in detail nor tested anything. So once the comments are addressed (or dismisased ;-) ), feel free to merge.

Summing up individual comments, there are:

- Replacing 'BaseException' with more specififc exceptions.
- Using _single_underscore instead of __double_underscore
- Changes from 'value' to e.g. 'transmission'. Is this policy, or is it just for now (which would be OK, actually).

**Comment by IvarsKarpics** [comment]
Thank you @rhfogh for the review. I will update PR with your suggested changes.

File: gui/bricks/AttenuatorsBrick.py (hunk 1 above)
Comment (IvarsKarpics) [reply gui/bricks/AttenuatorsBrick.py:138]: yes

File: gui/widgets/create_task_base.py (hunk 4 above)
Comment (IvarsKarpics) [reply gui/widgets/create_task_base.py:82]: no, not yet. We agreed to discuss signals during the next dev meeting

File: gui/widgets/heat_map_widget.py (hunk 5 above)
Comment (IvarsKarpics) [reply gui/widgets/heat_map_widget.py]: No, there is no need to hide the attribute. Changed to the single underscore.

File: gui/widgets/xray_imaging_results_widget.py (hunk 8 above)
Comment (IvarsKarpics) [reply gui/widgets/xray_imaging_results_widget.py]: last_image_button_clicked needs a class attribute, so it needs to stay in the class. Well, it was a suggested by pylint to move methods outside the class if they do not use class attributes (which is ok and allows to use functions outside the class)

Comment (rhfogh) [reply gui/widgets/xray_imaging_results_widget.py]: Cool.

File: gui/widgets/create_task_base.py (hunk 4 above)
Comment (rhfogh) [reply gui/widgets/create_task_base.py:82]: Cool.

**Review by rhfogh — APPROVED** [review]
Please feel free to merge (unless you want comments from others first).

**Pull request merged**